assets/javascripts/discourse/components/query-result.js.es6

```diff
@@ -139,6 +140,61 @@ const QueryResultComponent = Ember.Component.extend({
     return transformedRelTable(groups);
   },
 
+  @computed(
+    "rows.[]",
+    "content.colrender.[]",
+    "content.result_count",
+    "colCount"
+  )
+  canShowGraph(rows, colrender, result_count, colCount) {
+    const hasTwoColumns = colCount === 2;
+    const secondColumnContainsNumber =
+      result_count > 0 && typeof rows[0][1] === "number";
+    const secondColumnContainsId = colrender[1] !== undefined;
+
+    return (
+      hasTwoColumns && secondColumnContainsNumber && !secondColumnContainsId
+    );
+  },
+
+  @computed("content.rows.[]", "content.colrender.[]")
+  graphLabels(rows, colRender) {
+    const labelSelectors = new Map([
```

Comment:
could you explain the need for a Map here ?

Reply:
I just needed a hash-like structure to save these selectors in an easy-to-read way and to access them. I've chosen `Map` arbitrarily. A plain object will work perfectly well here - the keys are strings, and we don't use any special methods of `Map`.  

Well, I think it'll be simpler with a plain object, and I would like to use it. Is this what concerns you here?

And I consciously allocated it on the function level, not on the object level, in favor of readability. Is this alright?

Reply:
Yes I think a js object would be enough, we don't use Map much unless the order matters